Add blog post about making API more efficient (#14)
digital-land/digital-land.github.io: content/blog-post/making-our-api-more-efficient.md

---

> title: "API Efficiency: How Lean Code Cut Energy Use by 99%"

Reviewer: ```suggestion
title: "Making our API more efficient for data consumers"
```
Slight tweak, as we can't easily estimate the energy savings. Better to talk about making things more efficient for users.

date: 2024-08-13
author: "Ben Hodgkiss"
status: beta

> section: Infrastructure

Reviewer: ```suggestion
section: Planning data
```

This one is misleading, it defines the section of the website a post falls in rather than the team.

summary: Working with PlanX, the Planning Data team have been able to streamline our API to boost efficiency and reduce energy.
---

The amount of electricity used by data centres is increasing rapidly. Recent research in the Republic of Ireland shows a [400% increase](https://datacentremagazine.com/critical-environments/power-hungry-data-centres-put-pressure-on-irelands-grid) from 2015 to 2024. As AI and digital services grow, it’s important that we aim to minimise the pressure on data centres by providing lean code with the minimal outputs that use the least electricity.

> ## The Challenge of Data Bloat

Reviewer: ```suggestion
## The challenge of data bloat
```
Small point on style, we use sentence case for headings.

However, it can be really hard to think about how to do that when you’re in the middle of a large project like [Planning Data](https://www.planning.data.gov.uk/about/). This service collects planning and housing data from a wide variety of sources and transforms it into a consistent state available to all. 

The platform reduces the work which every planning service needs to undertake to find and make planning data usable, and creates opportunities to learn from users on ways we can further reduce this duplicated effort.

> [PlanX](https://www.planx.uk/) are one of the services built on the Planning Data Platform. Their product uses the Planning Data API to check if a property is within an area of conservation, affected by an article 4 direction, has a high flood risk along with a number of other considerations offered by the API.

Reviewer: ```suggestion
[PlanX](https://www.planx.uk/) are one of the services built using data from the Planning Data platform. Their product uses [our API](https://www.planning.data.gov.uk/docs) to check if a property is within an area of conservation, affected by an Article 4 direction, or has a high flood risk along with a number of other considerations offered by the API.
```

> ## A Targeted Solution: The Exclude-Field Parameter

Reviewer: ```suggestion
## A targeted solution
```

When we heard this from PlanX, we quickly workshopped a few ideas and decided on introducing an `exclude-field` parameter to the API. This would allow users to do exactly what they wanted - bring back everything except those large geometry files.

Within 3 weeks, we released this back to users. Within 24 hours, PlanX had implemented the parameter across their queries.

> ## Efficiency & Environmental Benefits

Reviewer: ```suggestion
## Efficiency and benefits
```

> ## Lessons for Sustainable Software Development

Reviewer: ```suggestion
## Lessons for sustainable software development
```
